src/target/ext/base/multithread.rs

```diff
@@ -186,6 +193,23 @@ pub trait MultiThreadOps: Target {
     }
 }
 
+/// Target Extension - [Reverse execution] for multi threaded targets.
+///
+/// Reverse execution allows the target to run backwards up until there is
+/// no more execution history.
+///
+/// [Reverse execution]: https://sourceware.org/gdb/current/onlinedocs/gdb/Reverse-Execution.html
+pub trait MultiThreadReverseStep: Target + MultiThreadOps {
+    /// Reverse-step the specified [Tid] by one instruction.
+    fn reverse_step(
```

Comment:
Is there a reason you're splitting `reverse_step` and `reverse_continue` into two separate functions (as opposed to using that `ReverseResumeAction` enum I mentioned earlier)? Or have you just not gotten around to updating it yet?

Reply:
It's just because the GDB protocol also has them split into `ReverseContinue+;` and `ReverseStep+;` features. Do you think I should combine them anyways?

Reply:
Ohhh, true...

If possible, could you look in to whether the two features are coupled, as https://github.com/daniel5151/gdbstub/issues/47#issuecomment-822215249 suggests? This might be an unfortunate case of the protocol implying one thing, but the GDB implementation doing another... 

In any case, I agree that the API should reflect the underlying protocol features, but it would be good to check if there's a undocumented dependency between the two, as that's something I would most certainly want to include in the documentation.

Reply:
I checked this again, and I'm now fairly sure you can separate out the features -- Not sure why I previously thought they were dependent on each other.